### PR TITLE
[`digital-carbon`] Combine async retire entities

### PR DESCRIPTION
**Pull request by psparacino**
This PR combines the `ToucanBridgeRequest` and `C3RetireRequest` into one entity `AsyncRetireRequest`. 

Any registry specific fields that aren't required for other async requests are saved in their own entity on `AsyncRetireRequest`.

### CODE REVIEW

File: polygon-digital-carbon/src/TransferHandler.ts
Comment (cujowolf) [inline polygon-digital-carbon/src/TransferHandler.ts:106]: Since the `Retire` entity is now non-nullable, should this be un-commented and the status updated? On the current test deployment the Puro retirement requested via tx hash 0xdbe415ed2b4e1e86e7a545c753e4f9dc2d325fa1107ec282cf2017f262b8fcb3 has a Finalized status under `asyncRetireRequest`, and a Requested status under `Retire`

File: polygon-digital-carbon/src/RetirementHandler.ts
Comment (cujowolf) [inline polygon-digital-carbon/src/RetirementHandler.ts:133]: Possible ID clashes if multiple registries start from 1 numerically? Puro is only passing the request ID while `getC3RetireRequestId` returns the combination of token + index.

**Review by cujowolf — CHANGES_REQUESTED** [review]
Need to also add the bridgeID value for Puro retirements at line 122 in RetirementHandler.ts. Currently returning a null value.

File: polygon-digital-carbon/src/TransferHandler.ts
Comment (psparacino) [reply polygon-digital-carbon/src/TransferHandler.ts]: fixed

File: polygon-digital-carbon/src/RetirementHandler.ts
Comment (psparacino) [reply polygon-digital-carbon/src/RetirementHandler.ts]: Combined asyncRetireRequest id generation into `createAsyncRetireRequestId` for both c3 and toucan, which takes the token address and index

**Review by 0xAurelius — APPROVED** [review]
LGTM but let's wait for Cujo to re-review before merging

**Review by cujowolf — APPROVED** [review]
Changes lgtm

**Pull request merged**
